tests/test_expectations.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2018 Carsten Blank
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for :mod:`pennylane_qiskit` expectation values
+"""
+import logging as log
+from pennylane import numpy as np
+
+from defaults import pennylane as qml, BaseTest
+from pennylane_qiskit import BasicAerQiskitDevice, AerQiskitDevice, IbmQQiskitDevice
+import unittest
+
+log.getLogger('defaults')
+
+
+class TestQVMBasic(BaseTest):
+    """Unit tests for the QVM simulator."""
+    # pylint: disable=protected-access
+
+    num_subsystems = 2
+    shots = 16 * 1024
+    ibmq_shots = 8 * 1024
+    devices = None
+
+    def setUp(self):
+        super().setUp()
+
+        self.devices = []
+        if self.args.device == 'basicaer' or self.args.device == 'all':
+            self.devices.append(BasicAerQiskitDevice(wires=self.num_subsystems, shots=self.shots))
+        if self.args.device == 'aer' or self.args.device == 'all':
+            self.devices.append(AerQiskitDevice(wires=self.num_subsystems, shots=self.shots))
+        if self.args.device == 'ibmq' or self.args.device == 'all':
+            if self.args.ibmqx_token is not None:
+                self.devices.append(
+                    IbmQQiskitDevice(wires=self.num_subsystems, shots=self.ibmq_shots, ibmqx_token=self.args.ibmqx_token))
+            else:
+                log.warning("Skipping test of the IbmQQiskitDevice device because IBM login credentials could not be "
+                            "found in the PennyLane configuration file.")
+
+    def test_identity_expectation(self):
+        """Test that identity expectation value (i.e. the trace) is 1"""
+        theta = 0.432
+        phi = 0.123
+        for dev in self.devices:
+            dev.apply('RX', wires=[0], par=[theta])
+            dev.apply('RX', wires=[1], par=[phi])
+            dev.apply('CNOT', wires=[0, 1], par=[])
+
+            O = qml.expval.qubit.Identity
+            name = 'Identity'
+
+            dev._expval_queue = [O(wires=[0], do_queue=False), O(wires=[1], do_queue=False)]
```

Comment:
The queue for `expval` for the device seems to be `expval_queue` and not `_expval_queue`. I do not understand this override.